repository.go

```diff
@@ -56,3 +60,26 @@ func Decode(object interface{}, data []byte) error {
 	}
 	return nil
 }
+
+func repoUpdate[T DomainObject](repo Repository[T], objectID string, object T, nbRetries int, updater func(T) (T, error)) (T, error) {
+	if nbRetries < 0 {
+		return object, errors.New("negative number of retries")
+	}
+
+	var err error
+	for i := 0; i <= nbRetries; i++ {
+		err = repo.Load(objectID, object)
+		if err != nil {
+			return object, err
+		}
+		object, err = updater(object)
+		if err != nil {
+			return object, err
+		}
+		err = repo.Save(object)
+		if err == nil || !errors.Is(err, ConcurrencyError) {
+			return object, err
+		}
```

Comment:
```suggestion
		if errors.Is(err, ConcurrencyError) {
			 continue
		}
		return object, err
```

Juste pour la lisibilité

Reply:
Bien vu